xilem/examples/gallery_label_xilem_view.rs

```diff
@@ -0,0 +1,348 @@
+    Axis, GridExt, Label, Padding, Prose, button, flex, grid, label, portal, prose, sized_box,
+};
+use xilem::{EventLoop, FontWeight, LineBreaking, TextAlignment, WidgetView, Xilem, palette::css};
+
+const LABEL_COLOR: Color = css::ROYAL_BLUE;
+
+struct AppState {
+    l1i1: TextAlignment,
+    l1i2: TextAlignment,
+    l1i3: TextAlignment,
+    l1i4: TextAlignment,
+    l1i5: TextAlignment,
+    l1i6: TextAlignment,
+    l2i1: TextAlignment,
+    l2i2: TextAlignment,
+    l2i3: TextAlignment,
+    l2i4: TextAlignment,
+    l2i5: TextAlignment,
+    l2i6: TextAlignment,
+    l3i1: TextAlignment,
+    l3i2: TextAlignment,
+    l3i3: TextAlignment,
+    l3i4: TextAlignment,
+    l3i5: TextAlignment,
+    l3i6: TextAlignment,
+    l4i1: TextAlignment,
+    l4i2: TextAlignment,
+    l4i3: TextAlignment,
+    l4i4: TextAlignment,
+    l4i5: TextAlignment,
+    l4i6: TextAlignment,
+    l4i1x: CrossAxisAlignment,
+}
+impl Default for AppState {
+    fn default() -> Self {
+        Self {
+            l1i1: TextAlignment::Start,
+            l1i2: TextAlignment::Left,
+            l1i3: TextAlignment::Middle,
+            l1i4: TextAlignment::End,
+            l1i5: TextAlignment::Right,
+            l1i6: TextAlignment::Justified,
+            l2i1: TextAlignment::Start,
+            l2i2: TextAlignment::Left,
+            l2i3: TextAlignment::Middle,
+            l2i4: TextAlignment::End,
+            l2i5: TextAlignment::Right,
+            l2i6: TextAlignment::Justified,
+            l3i1: TextAlignment::Start,
+            l3i2: TextAlignment::Left,
+            l3i3: TextAlignment::Middle,
+            l3i4: TextAlignment::End,
+            l3i5: TextAlignment::Right,
+            l3i6: TextAlignment::Justified,
+            l4i1: TextAlignment::Start,
+            l4i2: TextAlignment::Left,
+            l4i3: TextAlignment::Middle,
+            l4i4: TextAlignment::End,
+            l4i5: TextAlignment::Right,
+            l4i6: TextAlignment::Justified,
+            l4i1x: CrossAxisAlignment::Start,
+        }
+    }
+}
+fn text_align_cycle(cur: &TextAlignment) -> TextAlignment {
+    match cur {
+        TextAlignment::Start => TextAlignment::Left,
+        TextAlignment::Left => TextAlignment::Middle,
+        TextAlignment::Middle => TextAlignment::End,
+        TextAlignment::End => TextAlignment::Right,
+        TextAlignment::Right => TextAlignment::Justified,
+        TextAlignment::Justified => TextAlignment::Start,
+    }
+}
+fn text_x_align_cycle(cur: &CrossAxisAlignment) -> CrossAxisAlignment {
+    match cur {
+        CrossAxisAlignment::Start => CrossAxisAlignment::Center,
+        CrossAxisAlignment::Center => CrossAxisAlignment::End,
+        CrossAxisAlignment::End => CrossAxisAlignment::Baseline,
+        CrossAxisAlignment::Baseline => CrossAxisAlignment::Fill,
+        CrossAxisAlignment::Fill => CrossAxisAlignment::Start,
+    }
+}
+
+impl AppState {
+    fn realign(&mut self, pos: impl AsRef<str> + std::fmt::Debug) {
```

Comment:
I don't think that this is a great signature; if you really think that repeating `it.x = cycle(it.x)` is too much repetition, then maybe we can try `cycle(&mut it.x)`.
Stringly typed field access is not generally idiomatic in Rust code.

Reply:
You're right, it's not too much repetition, and the strings are bad, updated to remove them